### ISSUE TITLE
Migrate to styled-components 4

### ISSUE THREAD

**Issue by CarloPalinckx** [MyOnlineStore/bricks#182]
### Is your feature request related to a problem? Please describe ⚡️
styled-components just released v4 https://t.co/Qb1JwzuSwx which contains a few goodies for us to utilize.

### Describe the solution you'd like ✨
The `as` prop would allow us to skip a lot of boilerplate in a few components, namely:
- Box
- Button
- Heading
- Link
- Text

It would also be preferable to expose this on the `Box`. This could potentially solve a lot of html sematic issues (table elements for instance)

Native support for `ref`.

The demo shows this example:
```js
// Later in your render function
<Component ref={element => { this.myRef = element; }}
```

Not sure if refObjects are supported yet. Would be a nice opportunity to clean up some components.

**Comment by hborghols** [comment]
Looks sweet! The `as` prop and native `ref` support are awesome. I'm in favor.

**Comment by hborghols** [comment]
Maybe we can do this at the same time as #200 ? I do see that v4 is still in beta. Do we want to wait for the stable release or do we trust the styled-components team enough to migrate already/

**Comment by hborghols** [comment]
I don't think there is anything that requires the updates to be done at the same time so I'm in favor of waiting for a stable release of SC and in the meanwhile already go ahead with updating React.

**Comment by hborghols** [comment]
Stable 🎊:
https://www.styled-components.com/docs/faqs#what-do-i-need-to-do-to-migrate-to-v4

**Comment by CarloPalinckx** [comment]
fixed with #318
